pkg/controller/filter.go

```diff
@@ -35,6 +38,24 @@ type Filter interface {
 type MastershipFilter struct {
 	Store    mastershipstore.Store
 	Resolver DeviceResolver
+	nodeID   cluster.NodeID
+}
+
+// NewMastershipFilter creates an instance of mastershipFilter
+func NewMastershipFilter(store mastershipstore.Store, resolver DeviceResolver, nodeID cluster.NodeID) *MastershipFilter {
```

Comment:
Actually, I think we should not have a `New*` function for the mastership filter. We should keep the API the same, but use `cluster.GetNodeID()` if the private `nodeID` field is empty.

Reply:
@kuujo 
Done.